src/lib.rs

```diff
@@ -124,7 +124,81 @@ mod functional_test;
 #[macro_use]
 mod macros;
 
-pub use chrono;
+/// Re-export of the `time` crate
+///
+/// Tantivy uses [`time`](https://crates.io/crates/time) for dates.
+pub use time;
+
+use crate::time::{OffsetDateTime, PrimitiveDateTime, UtcOffset};
+
+/// A date/time value with second precision.
+///
+/// This timestamp does not carry any time zone information.
+/// Users are responsible for applying the provided conversion
+/// functions consistently.
+///
+/// All constructors and conversions are provided as explicit
+/// functions and not by implementing any `From`/`Into` traits
+/// to prevent unintended usage.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct DateTime {
```

Comment:
Can we have a nicer implem of Debug showing Rfc3339?

Reply:
I wonder if we should rename it `UtcDateTime`?

Reply:
Just the date is ok I think.

Reply:
https://github.com/quickwit-oss/tantivy/pull/1307/commits/d6b46a3da14672fc9f2429115c53c06ce6f20f43

The documentation of the `DateTime` wrapper now mentions the internal UTC representation, because that leaks not only for debugging but also when serializing as a JSON value.

I have noticed that the query parser is not tested with explicit time zone offsets, only with UTC? But that's unrelated and a different task.